### ISSUE TITLE
How to edit css / styling?

### ISSUE THREAD

**Issue by laura-amalie-graphics** [konstantinmuenster/gatsby-theme-portfolio-minimal#34]
Hi there,
I am a ux designer dabbling into frontend development for fun. A step on the way is a Gatsby portfolio. 
However, I cannot for the life of me figure out where I can edit colors, fonts, etc.? I tried adding the globalStyles but I could not get it to work....

Can anyone/you help?

**Comment by konstantinmuenster** [comment]
Hey @laura-amalie-graphics !

Have you tried [this](https://github.com/konstantinmuenster/gatsby-theme-portfolio-minimal/tree/main/gatsby-theme-portfolio-minimal#changing-the-color-scheme)?

**Comment by laura-amalie-graphics** [comment]
Yes I have - but I cannot get it to work. Do I need to import it somewhere? I tried adding the file you linked to, but whenever I tried changing fx the background color, nothing happened.

**Comment by konstantinmuenster** [comment]
No problem! You can try to delete the following steps:

1. delete your `node_modules` directory
2. run `gatsby clean`
3. install the dependencies again `npm install`
4. try if your `theme.css` file gets picked up now in development mode

Hopefully that does the trick 🤞

**Comment by VirtualVasquez** [comment]
> No problem! You can try to delete the following steps:
> 
> 1. delete your `node_modules` directory
> 2. run `gatsby clean`
> 3. install the dependencies again `npm install`
> 4. try if your `theme.css` file gets picked up now in development mode
> 
> Hopefully that does the trick 🤞

I ran into this issue, and these steps fixed for me, although I had to run `npm install` first before running `gatsby clean`.

**Comment by tigerbytestudio** [comment]
I ran into this same issue and dug into the root cause — sharing here in case it helps others.

## Why [theme.css](cci:7://file:///Users/shenzong/repos/_tigerbyte/ian-hafkenschiel-blog/src/gatsby-theme-portfolio-minimal/globalStyles/theme.css:0:0-0:0) shadowing doesn't work

The documented approach (create [src/gatsby-theme-portfolio-minimal/globalStyles/theme.css](cci:7://file:///Users/shenzong/repos/_tigerbyte/ian-hafkenschiel-blog/src/gatsby-theme-portfolio-minimal/globalStyles/theme.css:0:0-0:0)) silently has no effect. The reason is in [Layout/index.tsx](cci:7://file:///Users/shenzong/repos/_tigerbyte/ian-hafkenschiel-blog/src/gatsby-theme-portfolio-minimal/components/Layout/index.tsx:0:0-0:0):

```ts
// gatsby-theme-portfolio-minimal/src/components/Layout/index.tsx
import '../../globalStyles/theme.css';
```

Gatsby's shadow resolver only intercepts files imported **through the theme's package name**. Relative imports within the theme's own source are resolved directly from `node_modules` — the shadow resolver never sees them. So the shadow file is ignored regardless of cache state.

Clearing `node_modules` and re-installing may appear to fix it in some cases, but it doesn't address the root cause.

## Working fix

Shadow the [Layout](cci:1://file:///Users/shenzong/repos/_tigerbyte/ian-hafkenschiel-blog/src/gatsby-theme-portfolio-minimal/components/Layout/index.tsx:21:0-58:1) component itself and import your color overrides **after** [theme.css](cci:7://file:///Users/shenzong/repos/_tigerbyte/ian-hafkenschiel-blog/src/gatsby-theme-portfolio-minimal/globalStyles/theme.css:0:0-0:0) within it:

**[src/gatsby-theme-portfolio-minimal/components/Layout/index.tsx](cci:7://file:///Users/shenzong/repos/_tigerbyte/ian-hafkenschiel-blog/src/gatsby-theme-portfolio-minimal/components/Layout/index.tsx:0:0-0:0)**
```ts
import '@fontsource/roboto/400.css';
import '@fontsource/roboto/700.css';
import '../../globalStyles/global.css';
import '../../globalStyles/theme.css';
import '../../../styles/themeOverride.css'; // ← your overrides, loads after
// ... rest of the original Layout component unchanged
```

**[src/styles/themeOverride.css](cci:7://file:///Users/shenzong/repos/_tigerbyte/ian-hafkenschiel-blog/src/styles/themeOverride.css:0:0-0:0)** — only declare what you want to change:
```css
html body[data-theme='lightTheme'] {
    --primary-color: #your-color;
    --background-color: #your-color;
    /* etc. */
}

html body[data-theme='darkTheme'] {
    --primary-color: #your-color;
    /* etc. */
}
```

Use `html body` instead of just `body` for slightly higher specificity as a safety net.

**Important**: restart `gatsby develop` after creating the new shadow file — new shadow files are not picked up by hot reload.

## Proposed upstream fix

Moving the [theme.css](cci:7://file:///Users/shenzong/repos/_tigerbyte/ian-hafkenschiel-blog/src/gatsby-theme-portfolio-minimal/globalStyles/theme.css:0:0-0:0) import from [Layout/index.tsx](cci:7://file:///Users/shenzong/repos/_tigerbyte/ian-hafkenschiel-blog/src/gatsby-theme-portfolio-minimal/components/Layout/index.tsx:0:0-0:0) into the theme's [gatsby-browser.js](cci:7://file:///Users/shenzong/repos/_tigerbyte/ian-hafkenschiel-blog/gatsby-browser.js:0:0-0:0) would make the documented shadowing approach work correctly, since [gatsby-browser.js](cci:7://file:///Users/shenzong/repos/_tigerbyte/ian-hafkenschiel-blog/gatsby-browser.js:0:0-0:0) files are merged in plugin order and the site's own runs last.

**Comment by tigerbytestudio** [comment]
I've submitted a fix for this in #60. It's a minimal two-line change — moves the `theme.css` import from `Layout/index.tsx` into `gatsby-browser.js` so Gatsby's shadow resolver can intercept it correctly.
